vignettes/adrs_imwg.Rmd

```diff
@@ -0,0 +1,902 @@
+
+In this case we use the overall response records from `RS` from the
+investigator as our starting point. 
+It is worth emphasizing again that responses are not confirmed. 
+Confirmed values will be derived after further pre-processing.
+
+The parameter details such as
+`PARAMCD`, `PARAM` etc will always be company-specific, but an example
+is shown below so that you can trace through how these records feed into
+the other parameter derivations.
+
+```{r}
+adrs <- adrs %>%
+  filter(RSEVAL == "INVESTIGATOR" & RSTESTCD == "OVRLRESP") %>%
+  mutate(
+    PARAMCD = "OVR",
+    PARAM = "Overall Response by Investigator",
+    PARCAT1 = "Investigator",
+    PARCAT2 = "IMWG"
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, VISIT, RSTESTCD, RSEVAL, PARAMCD, PARAM, PARCAT1, PARCAT2)
+)
+```
+
+### Partial Date Imputation and Deriving `ADT`, `ADTF`, `AVISIT` etc
+
+If your data collection allows for partial dates, you could apply a
+company-specific imputation rule at this stage when deriving `ADT`. For
+this example, here we impute missing day to last possible date.
+
+```{r}
+adrs <- adrs %>%
+  derive_vars_dt(
+    dtc = RSDTC,
+    new_vars_prefix = "A",
+    highest_imputation = "D",
+    date_imputation = "last"
+  ) %>%
+  derive_vars_dy(
+    reference_date = TRTSDT,
+    source_vars = exprs(ADT)
+  ) %>%
+  mutate(AVISIT = VISIT)
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, PARAMCD, VISIT, AVISIT, RSDTC, ADT, ADTF, ADY)
+)
+```
+
+### Derive `AVALC` and `AVAL`
+
+Here we populate `AVALC` and create the numeric version as `AVAL`
+(ordered from worst to best response, followed by `NE`). The `AVAL` values are not considered in
+the parameter derivations below, and so changing `AVAL` here would not change
+the result of those derivations. However, please note that the ordering of `AVAL` 
+will be used to determine `ANL01FL` in the subsequent step, ensure that the appropriate 
+`mode` is being set in the `admiral::derive_var_extreme_flag()`.
+
+IMWG ordering will be used or if you'd like to provide your own company-specific ordering here you
+could do this as follows:
+
+```{r}
+aval_resp_new <- function(arg) {
+  case_when(
+    arg == "NE" ~ 8,
+    arg == "sCR" ~ 7,
+    arg == "CR" ~ 6,
+    arg == "VGPR" ~ 5,
+    arg == "PR" ~ 4,
+    arg == "MR" ~ 3,
+    arg == "SD" ~ 2,
+    arg == "PD" ~ 1,
+    TRUE ~ NA_real_
+  )
+}
+
+adrs <- adrs %>%
+  mutate(
+    AVALC = RSSTRESC,
+    AVAL = aval_resp_new(AVALC)
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, PARAMCD, AVISIT, ADT, AVAL, AVALC)
+)
+```
+
+## Derive Confirmed Overall Response Parameter {#covr}
+
+Confirmation of response require two consecutive readings of applicable disease 
+parameters (biochemical analyses). These need to be two distinct samples and 
+usually is performed on the same day or at next scheduled visit.  
```

Comment:
From our source article:
"Documentation of response requires two consecutive readings of the applicable disease parameter (serum M-protein, urine M-protein, or serum FLC), performed at any time (no minimum interval is required, it can be done the same day); however, to confirm response or progressive disease, two discrete samples are required; testing cannot be based upon the splitting of a single sample"

Reply:
@starosto  By reading the sentence, I felt that the confirmation visit/sample taken usually on the same day, which I am not sure. Can we say like "No minimal time interval is required for the subsequent disease assessment, but a different sample is required for confirmation." we can discuss in the call with the broader team.

Reply:
updated